Cosimulation-feature-2 | Cleanup editor feature

PandoraQS opened the pull request:
- Changed Maestro WebAPI to Maestro CLI
- Removed BottomBar (currently commented)
- Fixed windows build
- Renamed maestro process to simProcess
- Removed MonacoEditor and
- Cleaned up editor configuration
- Removed multi-model fetch and route from sidebar

PandoraQS committed:
Revert "cleaned up the code from comments and unused code"

This reverts commit e9df490f4316ef56a1c5b09c35f1dbc8a0277ceb.

PandoraQS committed:
Merge branch 'development' of https://github.com/PandoraQS/into-cps-application into development

PandoraQS committed:
fixed serve port

PandoraQS committed:
removed coe for maestro and fixed start and build configurations

SimoneMicalizzi committed:
new error for missing java, including env path for windows

PandoraQS committed:
Merge remote-tracking branch 'upstream/development' into development

PandoraQS committed:
Merge remote-tracking branch 'upstream/development' into development

PandoraQS committed:
Converted maestro webapi 3.0 to maestro webapi 4.0

PandoraQS committed:
- Included multi-model.json fetch from folder in Sidebar
- New Monaco editor module + dependency for esbuild worker manager
- Editor GUI to be fixed and improved

PandoraQS committed:
Included maestro-version json

PandoraQS committed:
changed maestro version to CLI (jar-with-dependencies)

PandoraQS committed:
- Changed Maestro WebApi to Maestro CLI
- Removed BottomBar (currently commented)
- Fixed windows build
- Renamed maestroprocess to simProcess
TODO:
- Categorize Snackbar Messages

PandoraQS committed:
removed Editor feature for clean PR

prasadtalasila commented on package.json:
```diff
@@ -69,13 +72,15 @@
   "dependencies": {
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
+    "@monaco-editor/react": "^4.7.0",
```
You are not using Monaco-editor anymore. Please remove this and direct Monaco import dependency.

PandoraQS replied:
Monaco editor/react has been removed from `package.json`.

prasadtalasila commented on package.json:
```diff
@@ -69,13 +72,15 @@
+    "monaco-editor": "^0.52.2",
     "monocart-coverage-reports": "^2.11.1",
     "net": "^1.0.2",
     "node-fetch": "^3.3.2",
```
Is this still being used?

PandoraQS replied:
Monaco editor has been removed from `package.json`.

prasadtalasila commented on src/types/global.d.ts:
```diff
@@ -22,10 +24,11 @@ export interface ICosimulationAPI {
   getSimulationResult: (sessionId: string) => Promise<string>;
   addListener: (event: string, callback: (...args: unknown[]) => void) => void;
   removeListener: (event: string, callback: (...args: unknown[]) => void) => void;
+  removeMultiModelPathListener: (callback: (path: string) => void) => void;
```
How's this used?

PandoraQS replied:
`removeMultiModelPathListener` has been removed.

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,195 @@
-  SimulationStatus,
-  MaestroNotifications,
-} from '../utils/constants/cosimulation/statuses';
+import { spawn } from 'child_process';
```
Please use execa npm package. It is better than child process API.

PandoraQS replied:
- `execa` has been installed and `spawn` has been removed
- The whole simulation process now uses `execa`.

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,195 @@
-    handleError(error);
-    return null;
+function sendSimulationStatus(status: string): void {
+  const { mainWindow } = require('../main');
```
It would be best for this code not to know about election application. It's responsibility is to manage maestro. Other modules can use it in a meaningful way to take results from here and send to windows.

PandoraQS replied:
`webContents` has been removed from maestro and doesn't know about electron.

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,195 @@
-  message: string,
-  logType: 'maestro' | 'cosimulation',
-) {
+function safeWrite(stream: fs.WriteStream | null, message: string) {
```
execa does stream management so hopefully you don't have to.

PandoraQS replied:
`safeWrite` has been removed.

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,195 @@
-        success: false,
-        error: `Maestro JAR not found at ${maestroJarPath}.`,
-      };
+      sendNotification(`[Simulation Error]: Maestro JAR not found at ${maestroJarPath}`, 'error');
```
This can be thrown as an exception which can be handled in caller code.

PandoraQS replied:
Included exceptions to handle notifications and the latter work fine.
<img src ="https://github.com/user-attachments/assets/4a906d50-0da7-47f8-9ddc-d23a742f1fbb " width="600" heigth="500"/>

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,195 @@
 
     if (!fs.existsSync(tempMaestroJarPath)) {
       fs.copyFileSync(maestroJarPath, tempMaestroJarPath);
+      console.log(`[INFO]: Copied Maestro JAR to temp path: ${tempMaestroJarPath}`);
```
Please use Winston npm package for logging.

PandoraQS replied:
Changed every consolewith winston info, warn and error logging in backend files.

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,195 @@
-      );
-      return '';
-    }
+      isSimulationInProgress = false;
```
"is" prefix can be removed.

PandoraQS replied:
"is" prefix from the variable name has been removed.

prasadtalasila commented on src/cosimulation/simulationContext.ts:
```diff
@@ -6,10 +6,6 @@ const simulationContext: SimulationContext = {
-  return simulationContext.sessionId;
-};
-
 export const setSessionId = (id: string): void => {
```
Given that the CLI is being used, is this still being used?

PandoraQS replied:
`simulationContext` has been deleted, as it is not needed anymore.

prasadtalasila commented on src/main.ts:
```diff
@@ -43,36 +43,26 @@ ipcMain.handle('maestro', async (event, args): Promise<MaestroResponse> => {
+          return { success: false, error: SimulationStatus.SimulationAlreadyInProgress };
+        }
+      
+        isStartSimulationRunning = true;
```
again, please remove "is" prefix from the variable name.

PandoraQS replied:
"is" prefix from the variable name has been removed.

prasadtalasila reviewed (commented):
@PandoraQS thanks for the PR. Please see the suggestions.

PandoraQS committed:
- Removed monaco editor from package
- Removed removeMultiModelPathListener
- Removed SimulationContext (including sessionId)
- "is" prefixes have been removed from isSimulationInProgress
- Installed winston for logging
- Created utils/logger.ts
- Changed logging for maestro.ts to winston logging
- Included execa and simplified logging writing feature with it, removing SafeWrite function
- Removed electron functions (webContents) from maestro.ts and modifed start-simulation function in main.ts
- Exported SimulationStatus as type
- Fixed platform error in maestroUtils and main.ts

PandoraQS committed:
- Included Winston logging in main.ts, menu.ts, config.ts, window.ts, errorHandler.ts, maestroUtils.ts
- Fixed platform warning in menu.ts
- Fixed env warning in window.ts
- Fixed error simulation not triggering snackbar

PandoraQS commented:
Last two commits included these changes:
- Removed `removeMultiModelPathListener`
- Removed `SimulationContext` (including `sessionId`)
- "is" prefixes have been removed
- Installed Winston for logging
- Created `utils/logger.ts`
- Changed logging for `maestro.ts` to Winston logging
- Included `execa` and simplified logging writing feature with it, removing the `safeWrite` function
- Removed electron event emitter `webContents` from `maestro.ts` and modifed start-simulation function in `main.ts`
- Exported SimulationStatus as type
- Fixed platform error in `maestroUtils.ts` and `main.ts`
- Included Winston logging in `main.ts`, `menu.ts`, `config.ts`, `window.ts`, `errorHandler.ts`, `maestroUtils.ts`
- Fixed platform warning in `menu.ts`
- Fixed env warning in `window.ts`
- Fixed error simulation not triggering snackbar

PandoraQS committed:
Updated Simulation Guide

prasadtalasila commented on src/cosimulation/maestro.ts:
```diff
@@ -1,487 +1,164 @@
-    sendNotification(MaestroNotifications.Error.ConfigurationNotSet, 'error');
-    return null;
+import { setupSimulationLogger, logInfo, logError, logWarn } from '../utils/logger';
+import execa from 'execa';
```
is this package in package.json?

PandoraQS replied:
Included in new commit

prasadtalasila commented on src/electron/gui/menu.ts:
```diff
@@ -1,7 +1,9 @@
+import { logError } from '../../utils/logger';
 
 let cosimulationEnabled = false;
+const platform = (process as any).platform;
```
how is this variable used?

PandoraQS replied:
Platform is being used to determine the shortcut to run the CoSimulation, either 'Cmd+F2' for darwin or 'Alt+F2' otherwise.

prasadtalasila reviewed (commented):
@PandoraQS please see the comments. In addition, please add / update tests for the new code. Thanks.

PandoraQS committed:
Re-installed execa properly.

PandoraQS committed:
- Fixed Configuration
- Fixed Unit Testing Configuration
- Improved unit test coverage across utils and components
- Maestro (cosimulation) and main not fully covered

PandoraQS committed:
- Implemented Wrapper for Execa + new exported function in maestro.ts for testing reasons
- Fixed remaining unit tests
- Fixed warning on Jest transformer in jest config

PandoraQS committed:
Fixed not working playwright tests
